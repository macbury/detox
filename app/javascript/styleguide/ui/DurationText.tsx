import React, { useMemo } from 'react'
import Text from '../form/Text'

/**
 * Format the time from seconds to M:SS.
 * @param  {Number} secs Seconds to format.
 * @return {String}      Formatted time.
 */
function formatTime(secs : any) {
  let secNum = parseInt(secs, 10) // don't forget the second param
  let hours : any = Math.floor(secNum / 3600)
  let minutes : any = Math.floor((secNum - (hours * 3600)) / 60)
  let seconds : any = secNum - (hours * 3600) - (minutes * 60)

  if (hours < 10) {
    hours = "0"+hours
  }
  if (minutes < 10) {
    minutes = "0"+minutes
  }
  if (seconds < 10) {
    seconds = "0"+seconds
  }

  if (hours === '00') {
    return minutes + ':' + seconds
  } else {
    return hours + ':' + minutes + ':' + seconds
  }
}

export interface IDurationTextProps {
  children: React.ReactNode
}

export default function DurationText({ children, ...props } : IDurationTextProps) {
  const time = useMemo(() => {
    if (children >= 0) {
      return formatTime(children as number / 1000)
    } else {
      return '--:--'
    }
  }, [children])

  return <Text {...props}>{time}</Text>
}