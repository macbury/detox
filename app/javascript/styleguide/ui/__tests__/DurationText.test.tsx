import React from 'react'
import { shallow } from 'enzyme'
import snapshot from 'enzyme-to-json'

import DurationText from '../DurationText'

it('renders 00:10:00', () => {
  const component = shallow(<DurationText>{600000}</DurationText>)
  expect(snapshot(component)).toMatchSnapshot()
})

it('renders 01:10:00', () => {
  const component = shallow(<DurationText>{6600000}</DurationText>)
  expect(snapshot(component)).toMatchSnapshot()
})

it('renders 00:00:05', () => {
  const component = shallow(<DurationText>{5000}</DurationText>)
  expect(snapshot(component)).toMatchSnapshot()
})