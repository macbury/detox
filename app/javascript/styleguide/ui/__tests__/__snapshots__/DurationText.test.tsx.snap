// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 00:00:05 1`] = `
<Text>
  00:05
</Text>
`;

exports[`renders 00:10:00 1`] = `
<Text>
  10:00
</Text>
`;

exports[`renders 01:10:00 1`] = `
<Text>
  01:50:00
</Text>
`;
